discover/deployment/articles-dxp/01-installing-search-engine/06-xpack-security.markdown

```diff
@@ -0,0 +1,240 @@
+architecture of
+[X-Pack](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/configuring-security.html). 
+
+## Installing X-Pack [](id=installing-x-pack)
+
+1.  To 
+    [install X-Pack](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/installing-xpack-es.html)
+    and automatically grant it the required permissions (recommended), run
+
+        bin/elasticsearch-plugin install x-pack --batch
+
+    on each cluster node. The `--batch` option bypasses installation prompts for
+    granting permissions to X-Pack. 
+
+    You'll see log output detailing the permissions granted, finishing with
+    `Installed x-pack`:
+
+        -> Downloading x-pack from elastic
+        [=================================================] 100%   
+        @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+        @     WARNING: plugin requires additional permissions     @
+        @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+        * java.io.FilePermission \\.\pipe\* read,write
+        * java.lang.RuntimePermission accessClassInPackage.com.sun.activation.registries
+        * java.lang.RuntimePermission getClassLoader
+        * java.lang.RuntimePermission setContextClassLoader
+        * java.lang.RuntimePermission setFactory
+        * java.net.SocketPermission * connect,accept,resolve
+        * java.security.SecurityPermission createPolicy.JavaPolicy
+        * java.security.SecurityPermission getPolicy
+        * java.security.SecurityPermission putProviderProperty.BC
+        * java.security.SecurityPermission setPolicy
+        * java.util.PropertyPermission * read,write
+        See http://docs.oracle.com/javase/8/docs/technotes/guides/security/permissions.html
+        for descriptions of what these permissions allow and the associated risks.
+        @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+        @        WARNING: plugin forks a native controller        @
+        @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+        This plugin launches a native controller that is not subject to the Java
+        security manager nor to system call filters.
+        Elasticsearch keystore is required by plugin [x-pack], creating...
+        -> Installed x-pack
+
+    See more about the permissions X-Pack needs
+    [here](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/installing-xpack-es.html). 
+
+2.  Make sure Elasticsearch allows the automatic creation of indexes. If
+    you're unsure, check `elasticsearch.yml` for this property:
+
+        action.auto_create_index: false
+
+    This property is `true` by default, so if you don't see it in
+    `elasticsearch.yml`, there's nothing to worry about. See [Elastic's
+    documentation](https://www.elastic.co/guide/en/elasticsearch/reference/6.1/docs-index_.html#index-creation) for more information on automatic index creation.
+
+3.  Restart Elasticsearch.
```

Comment:
I'm not clear if we recommend shutting down ES before installing X-Pack. I did during my testing. However, I left portal running, and that blew up my log when I shut down ES to execute this Restart directive. Perhaps I should instruct users to shut portal down as well? Recommending a path that avoids log explosions is preferable.

Reply:
Okay, so it's portal that should remain shut down here during the installation. Should probably make that clear in the docs.

Reply:
Specific edits:
- Elasticsearch must be running to install x-pack, if memory serves
- Before restarting Elasticsearch portal must be shut down to avoid log exceptions.
This is reflective of this one portal node, one Elasticsearch node, local testing environment.

cc @lipusz, @arboliveira

Reply:
I believe Elasticsearch just needs to be running when modifying users (most notably, when we set built-in user passwords for X-Pack Security)

Reply:
@xbrianlee 
I can't find it searching my Elasticsearch log, but I believe ES needs to be running to install X-Pack. I think I got a failure message due to having the ES server shut down the first time.

Reply:
Nope, you're right--tested just now 
Here you go: :1st_place_medal: